lua/plenary/enum.lua

```diff
@@ -0,0 +1,30 @@
+local Enum = {}
+
+setmetatable(Enum, Enum)
```

Comment:
we should not do this because the class is not an enum, it is just a class that can create itself. If you want to give it the `__call` metatable to construct itself you should do it separately and not in enum. Maybe this will help https://github.com/nvim-lua/plenary.nvim/pull/134